### ISSUE TITLE
Linear.forward() takes 2 positional arguments but 3 were given

### ISSUE THREAD

**Issue by mrmuke** [InternLM/InternLM-XComposer#175]
I keep on getting Linear.forward() takes 2 positional arguments but 3 were given when running the finetune_lora.sh script

**Comment by mrmuke** [comment]
Hi, yes, here's a screenshot. 
<img width="966" alt="image" src="https://github.com/InternLM/InternLM-XComposer/assets/62921767/d3b8e5bb-6af4-45a8-bb14-c8e57010bcb9">

**Comment by mrmuke** [comment]
I also am at the latest version of the code

**Comment by mrmuke** [comment]
It seems to have to do with passing in im_mask in all the forward functions, when I removed it from all the parameters, it started working. Could this have to do with torch version or is it working for you guys?

**Comment by yuhangzang** [comment]
Thanks for your feedback. It looks like the original P-LoRA was replaced by LoRA.

**Comment by mrmuke** [comment]
How should this be fixed?

**Comment by yuhangzang** [comment]
I haven't encountered this problem in my env. Can you provide more details about your environment and package versions?

**Comment by mrmuke** [comment]
These are my torch and transformers versions:
torch==2.0.1
torchaudio==2.0.2
torchvision==0.15.2
transformers==4.33.2

**Comment by mrmuke** [comment]
I'm just running the finetune_lora.sh script

**Comment by mrmuke** [comment]
My full environment: CUDA 12.1

absl-py==2.0.0
accelerate==0.24.1
aiofiles==23.2.1
aiohttp==3.8.6
aiohttp-retry==2.8.3
aiosignal==1.3.1
altair==5.2.0
amqp==5.2.0
annotated-types==0.6.0
antlr4-python3-runtime==4.9.3
anyio==3.7.1
appdirs==1.4.4
asttokens==2.4.1
async-timeout==4.0.3
asyncssh==2.14.1
atpublic==4.0
attributedict==0.3.0
attrs==23.1.0
autoawq==0.1.7
Automat==20.2.0
Babel==2.8.0
bcrypt==3.2.0
billiard==4.2.0
bitsandbytes==0.41.2.post2
blessings==1.7
blinker==1.7.0
boto==2.49.0
boto3==1.29.6
botocore==1.32.6
CacheControl==0.12.10
cachetools==5.3.2
cachy==0.3.0
celery==5.3.5
certifi==2020.6.20
cffi==1.16.0
chardet==5.2.0
charset-normalizer==3.3.2
ci-info==0.3.0
cleo==0.8.1
click==8.1.7
click-didyoumean==0.3.0
click-plugins==1.1.1
click-repl==0.3.0
clikit==0.6.2
cloud-init==23.3.1
cmake==3.27.7
codecov==2.1.13
colorama==0.4.6
coloredlogs==15.0.1
colour-runner==0.1.1
comm==0.2.1
command-not-found==0.3
configobj==5.0.6
configparser==6.0.0
constantly==15.1.0
contourpy==1.2.0
coverage==7.3.2
crashtest==0.3.1
cryptography==41.0.5
cycler==0.12.1
dataclasses-json==0.6.2
DataProperty==1.0.1
datasets==2.17.0
dbus-python==1.2.18
debugpy==1.8.0
decorator==5.1.1
deepdiff==6.7.1
deepspeed==0.13.2
dictdiffer==0.9.0
dill==0.3.7
diskcache==5.6.3
distlib==0.3.7
distro==1.7.0
distro-info==1.1+ubuntu0.1
dnspython==2.4.2
docker==5.0.3
docker-compose==1.29.2
dockerpty==0.4.1
docopt==0.6.2
dpath==2.1.6
dulwich==0.21.6
dvc==3.28.0
dvc-data==2.20.0
dvc-http==2.30.2
dvc-objects==1.2.0
dvc-render==0.6.0
dvc-studio-client==0.15.0
dvc-task==0.3.0
dvclive==3.2.0
editdistance==0.6.2
einops==0.7.0
entrypoints==0.4
et-xmlfile==1.1.0
etelemetry==0.3.1
evaluate==0.4.1
exceptiongroup==1.1.3
executing==2.0.1
fastapi==0.104.1
ffmpy==0.3.2
filelock==3.13.1
fitz==0.0.1.dev2
Flask==3.0.0
Flask-Cors==4.0.0
flatten-dict==0.4.2
flufl.lock==7.1.1
fonttools==4.47.0
frozenlist==1.4.0
fsspec==2023.10.0
funcy==2.0
future==0.18.3
gitdb==4.0.11
GitPython==3.1.40
gradio==4.18.0
gradio_client==0.10.0
grandalf==0.8
greenlet==3.0.1
gto==1.5.0
gyp==0.1
h11==0.14.0
hjson==3.1.0
html5lib==1.1
httpcore==1.0.2
httplib2==0.20.2
httptools==0.6.1
httpx==0.25.1
huggingface-hub==0.20.3
humanfriendly==10.0
hvac==2.0.0
hydra-core==1.3.2
hyperlink==21.0.0
idna==3.3
importlib-metadata==4.6.4
importlib-resources==6.1.1
incremental==21.3.0
inspecta==0.1.3
iotop==0.6
ipykernel==6.29.0
ipython==8.20.0
isodate==0.6.1
iterative-telemetry==0.0.8
itsdangerous==2.1.2
jaro-winkler==2.0.3
jedi==0.19.1
jeepney==0.7.1
Jinja2==3.1.2
jmespath==1.0.1
joblib==1.3.2
jsonlines==4.0.0
jsonpatch==1.33
jsonpointer==2.0
jsonschema==3.2.0
jupyter_client==8.6.0
jupyter_core==5.7.1
keyring==23.5.0
kiwisolver==1.4.5
kombu==5.3.3
langchain==0.0.340
langserve==0.0.30
langsmith==0.0.66
launchpadlib==1.10.16
lazr.restfulclient==0.14.4
lazr.uri==1.0.6
Levenshtein==0.23.0
lit==17.0.4
lm-eval==0.3.0
lockfile==0.12.2
loguru==0.7.2
looseversion==1.3.0
lxml==4.9.4
markdown-it-py==3.0.0
MarkupSafe==2.1.3
marshmallow==3.20.1
matplotlib==3.8.2
matplotlib-inline==0.1.6
mbstrdecoder==1.1.3
mdurl==0.1.2
more-itertools==8.10.0
mpmath==1.3.0
msgpack==1.0.7
multidict==6.0.4
multiprocess==0.70.15
mypy-extensions==1.0.0
nest-asyncio==1.6.0
netifaces==0.11.0
networkx==3.2.1
nibabel==5.2.0
ninja==1.11.1.1
nipype==1.8.6
nltk==3.8.1
numexpr==2.8.7
numpy==1.26.2
nvidia-cublas-cu11==11.10.3.66
nvidia-cublas-cu12==12.1.3.1
nvidia-cuda-cupti-cu11==11.7.101
nvidia-cuda-cupti-cu12==12.1.105
nvidia-cuda-nvrtc-cu11==11.7.99
nvidia-cuda-nvrtc-cu12==12.1.105
nvidia-cuda-runtime-cu11==11.7.99
nvidia-cuda-runtime-cu12==12.1.105
nvidia-cudnn-cu11==8.5.0.96
nvidia-cudnn-cu12==8.9.2.26
nvidia-cufft-cu11==10.9.0.58
nvidia-cufft-cu12==11.0.2.54
nvidia-curand-cu11==10.2.10.91
nvidia-curand-cu12==10.3.2.106
nvidia-cusolver-cu11==11.4.0.1
nvidia-cusolver-cu12==11.4.5.107
nvidia-cusparse-cu11==11.7.4.91
nvidia-cusparse-cu12==12.1.0.106
nvidia-nccl-cu11==2.14.3
nvidia-nccl-cu12==2.19.3
nvidia-nvjitlink-cu12==12.3.101
nvidia-nvtx-cu11==11.7.91
nvidia-nvtx-cu12==12.1.105
oauthlib==3.2.0
omegaconf==2.3.0
openai==1.10.0
openpyxl==3.1.2
ordered-set==4.1.0
orjson==3.9.10
packaging==23.2
pandas==2.1.3
parso==0.8.3
pastel==0.2.1
pathlib==1.0.1
pathspec==0.11.2
pathvalidate==3.2.0
pdf2image==1.16.3
peft==0.6.1
pexpect==4.8.0
Pillow==10.1.0
pkginfo==1.8.2
platformdirs==3.11.0
pluggy==1.3.0
poetry==1.1.12
poetry-core==1.0.7
portalocker==2.8.2
prompt-toolkit==3.0.43
protobuf==4.25.2
prov==2.0.0
psutil==5.9.6
ptyprocess==0.7.0
pure-eval==0.2.2
py-cpuinfo==9.0.0
pyarrow==14.0.1
pyarrow-hotfix==0.6
pyasn1==0.4.8
pyasn1-modules==0.2.1
pybind11==2.11.1
pycocotools==2.0.7
pycountry==22.3.5
pycparser==2.21
pydantic==2.6.1
pydantic_core==2.16.2
pydot==1.4.2
pydub==0.25.1
pygit2==1.13.2
Pygments==2.16.1
PyGObject==3.42.1
pygtrie==2.5.0
PyHamcrest==2.0.2
PyJWT==2.3.0
pylev==1.2.0
pymongo==4.6.1
PyMuPDF==1.23.8
PyMuPDFb==1.23.7
pynvml==11.5.0
pyOpenSSL==23.3.0
pyparsing==2.4.7
PyPDF2==3.0.1
pyproject-api==1.6.1
pyrsistent==0.18.1
pyserial==3.5
pytablewriter==1.2.0
pytesseract==0.3.10
python-apt==2.4.0+ubuntu2
python-dateutil==2.8.2
python-debian==0.1.43+ubuntu1.1
python-dotenv==1.0.0
python-Levenshtein==0.23.0
python-magic==0.4.24
python-multipart==0.0.6
pytz==2022.1
pyxnat==1.6
PyYAML==5.4.1
pyzmq==25.1.2
rapidfuzz==3.6.1
ray==2.8.0
rdflib==7.0.0
regex==2023.10.3
requests==2.31.0
requests-toolbelt==0.9.1
responses==0.18.0
rich==13.6.0
rootpath==0.1.1
rouge-score==0.1.2
ruamel.yaml==0.18.5
ruamel.yaml.clib==0.2.8
ruff==0.2.1
s3transfer==0.7.0
sacrebleu==1.5.0
safetensors==0.4.2
scikit-learn==1.3.2
scipy==1.11.3
scmrepo==1.4.1
screen-resolution-extra==0.0.0
SecretStorage==3.3.1
semantic-version==2.10.0
semver==3.0.2
sentencepiece==0.1.99
service-identity==18.1.0
shellingham==1.4.0
shortuuid==1.0.11
shtab==1.6.4
simplejson==3.19.2
six==1.16.0
smmap==5.0.1
sniffio==1.3.0
sos==4.5.6
SQLAlchemy==2.0.23
sqlitedict==2.1.0
sqltrie==0.8.0
ssh-import-id==5.11
stack-data==0.6.3
starlette==0.27.0
sympy==1.12
systemd-python==234
tabledata==1.3.3
tabulate==0.9.0
tcolorpy==0.1.4
tenacity==8.2.3
termcolor==2.4.0
tesserocr==2.6.2
texttable==1.6.4
threadpoolctl==3.2.0
timm==0.9.12
tokenizers==0.13.3
toml==0.10.2
tomli==2.0.1
tomlkit==0.12.0
toolz==0.12.1
torch==2.0.1
torchaudio==2.0.2
torchvision==0.15.2
tornado==6.4
tox==4.11.4
tqdm==4.66.1
tqdm-multiprocess==0.0.11
traitlets==5.14.1
traits==6.3.2
transformers==4.33.2
triton==2.0.0
Twisted==22.1.0
typepy==1.3.2
typer==0.9.0
typing-inspect==0.9.0
typing_extensions==4.8.0
tzdata==2023.3
ubuntu-advantage-tools==8001
ubuntu-drivers-common==0.0.0
ufw==0.36.1
unattended-upgrades==0.1
urllib3==1.26.5
uvicorn==0.24.0.post1
uvloop==0.19.0
vine==5.1.0
virtualenv==20.25.0
vllm==0.2.2
voluptuous==0.14.0
wadllib==1.3.6
watchfiles==0.21.0
wcwidth==0.2.9
webencodings==0.5.1
websocket-client==1.2.3
websockets==11.0.3
Werkzeug==3.0.1
xformers==0.0.22.post7
xkit==0.0.0
xxhash==3.4.1
yarl==1.9.2
zc.lockfile==3.0.post1
zipp==1.0.0
zope.interface==5.4.0
zstandard==0.22.0

**Comment by yuhangzang** [comment]
Can u try `peft==0.8.2` when u are available and see if the issue persists?

**Comment by mrmuke** [comment]
It seems to be working now! Thanks!
